src/window.rs

```diff
@@ -156,6 +158,36 @@ impl Window {
             .or_else(|_| create_gl_window(window_builder, &event_loop, true))?;
         window.show();
 
+        #[cfg(not(any(windows, target_os = "macos")))]
+        {
+            if window_config.decorations() == Decorations::Transparent && event_loop.is_wayland() {
+                let bg = colors.primary.background;
+                let mut scale = 382.5 / (f32::from(bg.r) + f32::from(bg.g) + f32::from(bg.b));
+                scale /= 3.;
+                let scaled_bg = [0xFF, (f32::from(bg.r) * scale) as u8, (f32::from(bg.g) * scale) as u8, (f32::from(bg.b) * scale) as u8];
+                scale *= 1.5;
+                let vscaled_bg = [0xFF, (f32::from(bg.r) * scale) as u8, (f32::from(bg.g) * scale) as u8, (f32::from(bg.b) * scale) as u8];
+
+                let fg = colors.primary.foreground;
+                let red = colors.normal.red;
+                let green = colors.normal.green;
+                let blue = colors.normal.blue;
+
+                window.window().set_wayland_theme(WaylandTheme {
+                    primary_active: scaled_bg,
+                    primary_inactive: vscaled_bg,
+                    secondary_active: [0xFF, fg.r, fg.g, fg.b],
+                    secondary_inactive: [0x00, 0x00, 0x00, 0x00],
+                    close_button_hovered: [0xFF, red.r, red.g, red.b],
+                    close_button: [0x00, 0x00, 0x00, 0x00],
+                    maximize_button_hovered: [0xFF, green.r, green.g, green.b],
+                    maximize_button: [0x00, 0x00, 0x00, 0x00],
+                    minimize_button_hovered: [0xFF, blue.r, blue.g, blue.b],
+                    minimize_button: [0x00, 0x00, 0x00, 0x00],
```

Comment:
These are all mixed up I think. Testing on weston, the first three fields are RGB, not the last three.

Reply:
Actually, looks like they are BGR.

Reply:
Ah welcome to another ‘feature’ of wayland.

Pixels written to the buffer must be of native endian (little endian on Linux). Technically two versions of the colours should be supplied and used according to the system endianness.

Reply:
There should be code out there to switch the endianness to system endian, or it would be trivial to do for some custom type.

Reply:
Yeah there is. SCTK used to hardcode for performance however because this would only be called once I think you’re right that a trivial utility function would be best.